src/devices/vdml_adi.cpp

```diff
@@ -211,6 +211,33 @@ double Potentiometer::get_angle() const {
 	return ext_adi_potentiometer_get_angle(merge_adi_ports(temp_smart, _adi_port));
 }
 
+Pneumatics::Pneumatics(std::uint8_t adi_port, bool initial_state) 
+: DigitalOut(adi_port), initial_state(initial_state), state(initial_state) {
+}
+
+Pneumatics::Pneumatics(ext_adi_port_pair_t port_pair, bool initial_state) 
+: DigitalOut(std::get<1>(port_pair)), initial_state(initial_state), state(initial_state) {
```

Comment:
This does not allow users to use the pneumatics with the ADI port expander. This should just call the constructor for the DigitalOut by passing port_pair rather than getting the 2nd element in the pair.